src/main/java/com/limechain/network/ConnectionManager.java

```diff
@@ -0,0 +1,50 @@
+package com.limechain.network;
+
+import com.limechain.network.protocol.blockannounce.scale.BlockAnnounceHandshake;
+import com.limechain.network.protocol.blockannounce.scale.BlockAnnounceMessage;
+import io.emeraldpay.polkaj.types.Hash256;
+import io.libp2p.core.PeerId;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class ConnectionManager {
+    private static ConnectionManager INSTANCE;
+    protected final Map<PeerId, BlockAnnounceHandshake> peers = new HashMap<>();
```

Comment:
Handshakes are one time messages and shouldn't be updated.  I'd rather create a new class like `PeerInfo` instead of using `BlockAnnounceHandshake` even if it contains the same fields in the begginning. In the future it will probably contain other fields like `reputation` and others